File: MyProjects/MAP/Assignment 7/Interpreter/Model/Utilities/IStack.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;

namespace Interpreter.Model.Utilities
{
    public interface IStack<T>:IEnumerable
    {
        void add(T x);
        List<T> getElements();
        bool isEmpty();
        T top();
        T pop();
    }
}

File: MyProjects/MAP/Practical Exam/Procedures/src/Model/Utilities/IStack.java
package Model.Utilities;

import Model.Statements.IStmt;

import java.util.List;

public interface IStack<T> {
    void add(T x);
    Iterable<T> getElements();
    boolean isEmpty();
    public List<T> getToList();
    T top();
    T pop();
    IStack<T> copy();
}
